src/NCronJob/Registry/JobRegistry.cs

```diff
@@ -167,4 +171,37 @@ public int GetHashCode(JobDefinition obj) => HashCode.Combine(
             obj.CustomName,
             obj.IsStartupJob);
     }
+
+    private sealed class DependentJobDefinitionEqualityComparer : IEqualityComparer<JobDefinition>
+    {
+        // TODO: Maybe is the code conflating two different concepts.
+        // Dependent jobs may have a name, a type and a parameter, but that's the most of it.
+        // And the code currently uses the same type to hold the configuration of "lead" jobs
+        // and dependent jobs.
+        //
+        // Which brings this dependent job only comparer.
+        //
+        // Maybe should a DependentJobDefinition type spawn?
+
+        public static readonly DependentJobDefinitionEqualityComparer Instance = new();
+
+        public bool Equals(JobDefinition? x, JobDefinition? y) =>
+            (x is null && y is null) || (x is not null && y is not null
+                                         && x.Type == y.Type && x.Type != typeof(DynamicJobFactory)
+                                         && x.Parameter == y.Parameter
+                                         //&& x.CronExpression == y.CronExpression
```

Comment:
If not needed, I would remove the commented out code.

Reply:
No problem. I left it commented for you to notice the change. I'll drop it with the next push.

Reply:
Fixed